examples/glium.rs

```diff
@@ -142,12 +156,37 @@ fn main() -> io::Result<()> {
                 }
                 _ => panic!("invalid buffer pixelformat"),
             };
+            let _ = event_loop_proxy.send_event(UserEvent::WakeUp);
             tx.send(data).unwrap();
         }
     });
+    
+    struct LoopHandler<F> {
+        user_event: F,
+    }
+    
+    impl<F: Fn(UserEvent)> ApplicationHandler<UserEvent> for LoopHandler<F> {
+        fn resumed(&mut self, event_loop: &ActiveEventLoop) {}
+
+        fn window_event(
+            &mut self,
+            event_loop: &ActiveEventLoop,
+            window_id: WindowId,
+            event: WindowEvent,
+        ) {
+            // polling and handling the events received by the window
+            if let winit::event::WindowEvent::CloseRequested = event {
+                event_loop.exit();
+            }
+        }
 
-    event_loop
-        .run(move |event, elwt| {
+        fn user_event(&mut self, event_loop: &ActiveEventLoop, event: UserEvent) {
```

Comment:
Note that if you know a value can only have one variant, you can pattern-match on it in the function signature:

```suggestion
        fn user_event(&mut self, event_loop: &ActiveEventLoop, UserEvent::WakeUp: UserEvent) {
```

This even works to unpack a tuple or struct field if you pass `data` as part of `UserEvent`. Then, since there's only one event, you can also change `enum` to `struct`.

Reply:
I think using the signature as the trait defines it is more readable.
About keeping this as an enum, I think most real-world applications are going to have some form of an enum, so this should make it easier to integrate.

I'm willing to change both if it's required for merging, though.